### PR TITLE
Uncrustified vital/algo (2)

### PR DESCRIPTION
**Pull request by quantumech3**

### CODE REVIEW

File: vital/algo/detected_object_set_input.h [hunk 1]
```diff
@@ -43,12 +46,14 @@ class VITAL_ALGO_EXPORT detected_object_set_input
...
   /// Open a file of detection sets.
+
   /**
    * This method opens a detection set file for reading.
    *
    * \param filename Name of file to open
    *
-   * \throws kwiver::vital::path_not_exists Thrown when the given path does not exist.
+   * \throws kwiver::vital::path_not_exists Thrown when the given path does not
+   * exist.
```
Comment (mwoehlke-kitware) [inline vital/algo/detected_object_set_input.h:56]: So... Doxygen indentation is not well specified, but I do know this isn't right. Given it's also the wrong style (should be all `///`), do we even want to bother with this for the moment? (There are a bunch of spurious blank lines also being introduced due to the mix of `///` and `/**` that I'm ignoring for now.)

File: vital/algo/detected_object_set_input.h [hunk 2]
```diff
@@ -113,8 +123,13 @@ class VITAL_ALGO_EXPORT detected_object_set_input
...
-typedef std::shared_ptr<detected_object_set_input> detected_object_set_input_sptr;
+typedef std::shared_ptr< detected_object_set_input >
+  detected_object_set_input_sptr;
+
+} // namespace algo
+
+} // namespace vital
 
-} } } // end namespace
+}     // end namespace
```
Comment (mwoehlke-kitware) [inline vital/algo/detected_object_set_input.h:133]:
```suggestion
} // namespace kwiver
```
Other instances exist; I'm not going to individually flag them. (One of these days I'd really like to teach uncrustify to forcibly remove existing namespace comments...)

File: vital/algo/detected_object_set_output.cxx [hunk 3]
```diff
@@ -18,17 +18,19 @@
...
+
 namespace vital {
+
 namespace algo {
 
 detected_object_set_output
 ::detected_object_set_output()
   : m_stream( 0 )
-  , m_stream_owned( false )
+    , m_stream_owned( false )
```
Comment (mwoehlke-kitware) [inline vital/algo/detected_object_set_output.cxx:33]: This is somewhat atrocious. I doubt Uncrustify is correctly configured for this style, and anyway I'm not sure we should be encouraging it. (I also seem to recall Uncrustify is failing to move the `,` to the preceding line, despite theoretically being configured to do so; n.b. `pos_constr_comma`.)

File: vital/algo/detected_object_set_output.h [hunk 4]
```diff
@@ -35,28 +38,35 @@ namespace algo {
...
 public:
   virtual ~detected_object_set_output();
 
   /// Return the name of this algorithm
-  static std::string static_type_name() { return "detected_object_set_output"; }
+  static std::string
+  static_type_name()
+  {
+    return "detected_object_set_output";
+  }
```
Comment (mwoehlke-kitware) [inline vital/algo/detected_object_set_output.h:51]: Not the end of the world, but one of these days I'd really like to figure out why Uncrustify isn't permitting one-liners. Supposedly it is configured to do so.

File: vital/algo/dynamic_configuration.cxx [hunk 5]
```diff
@@ -2,11 +2,13 @@
 // OSI-approved BSD 3-Clause License. See top-level LICENSE file or
 // https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
 
-#include <vital/algo/dynamic_configuration.h>
 #include <vital/algo/algorithm.txx>
+#include <vital/algo/dynamic_configuration.h>
```
Comment (mwoehlke-kitware) [inline vital/algo/dynamic_configuration.cxx:6]: Directly corresponding headers (i.e. `foo.h`, `foo_p.h` in `foo.cxx`) should be first, and separated by a blank line from other headers. I see this elsewhere:
```suggestion
#include "dynamic_configuration.h"

#include <vital/algo/algorithm.txx>
```
Not sure what our convention is for `""` includes vs. `<>` includes for such headers, but in any case, please add the blank line.

File: vital/algo/estimate_fundamental_matrix.h [hunk 6]
```diff
@@ -12,66 +12,84 @@
...
 
-/// An abstract base class for estimating a fundamental matrix from matching 2D points
+/// An abstract base class for estimating a fundamental matrix from matching 2D
+/// points
 class VITAL_ALGO_EXPORT estimate_fundamental_matrix
-  : public kwiver::vital::algorithm_def<estimate_fundamental_matrix>
+  : public kwiver::vital::algorithm_def< estimate_fundamental_matrix >
 {
 public:
+
```
Comment (mwoehlke-kitware) [inline vital/algo/estimate_fundamental_matrix.h:35]: This looks like a bug. What Uncrustify are you using?

File: vital/algo/estimate_fundamental_matrix.h [hunk 7]
```diff
@@ -12,66 +12,84 @@
...
+  }
 
   /// Estimate an fundamental matrix from corresponding features
+
   /**
    * \param [in]  feat1 the set of all features from the first image
    * \param [in]  feat2 the set of all features from the second image
-   * \param [in]  matches the set of correspondences between \a feat1 and \a feat2
+   * \param [in]  matches the set of correspondences between \a feat1 and \a
+   * feat2
```
Comment (mwoehlke-kitware) [inline vital/algo/estimate_fundamental_matrix.h:49]: Prior comment about not fixing Doxygen stuff right now, I'm not actually sure if Doxygen like this...
```suggestion
   * \param [in]  matches the set of correspondences between \a feat1 and
   *                      \a feat2
```

**Review by mwoehlke-kitware — CHANGES_REQUESTED** [review]
This is just a start, but as there are issues that need to be resolved that may largely invalidate my efforts to keep going, I'm going to go ahead and submit what I have so far.

File: vital/algo/dynamic_configuration.cxx (hunk 5 above)
Comment (hdefazio) [reply vital/algo/dynamic_configuration.cxx:6]: @mwoehlke-kitware Did you ever determine what the "" vs <> convention is?

File: vital/algo/detected_object_set_output.h (hunk 4 above)
Comment (gbrooksby) [reply vital/algo/detected_object_set_output.h:51]: In this specific case, the "one liner" is ruled out by the constraint of 79 chars/line.  As a one-liner, this line comes to 80 characters, so Uncrustify is re-writing it.  I tested it with a setting of 80 chars/line and the one-line is retained.

File: vital/algo/dynamic_configuration.cxx (hunk 5 above)
Comment (gbrooksby) [reply vital/algo/dynamic_configuration.cxx:6]: I don't see a Kwiver convention mentioned [here](https://docs.google.com/document/d/1I5gP61K2VuV1Ki10uwpK0zd6o--4wR-uDtidl33ML0g/edit#) or [here](https://github.com/Kitware/kwiver/blob/master/doc/code-style.rst).  Perhaps there is a broader Kitware standard we could find and apply, but so far I'm not finding one.

File: vital/algo/estimate_fundamental_matrix.h (hunk 7 above)
Comment (gbrooksby) [reply vital/algo/estimate_fundamental_matrix.h:49]: I tried both formats and Doxygen doesn't care.  They both look the same and look good.  I agree however, that your suggestion is an improvement when looking at the source.

File: vital/algo/read_track_descriptor_set.h [hunk 8]
```diff
@@ -43,12 +46,14 @@ class VITAL_ALGO_EXPORT read_track_descriptor_set
...
   /// Open a file of track descriptor sets.
+
   /**
    * This method opens a track descriptor set file for reading.
    *
    * \param filename Name of file to open
    *
-   * \throws kwiver::vital::path_not_exists Thrown when the given path does not exist.
+   * \throws kwiver::vital::path_not_exists Thrown when the given path does not
+   * exist.
```
Comment (mwoehlke-kitware) [inline vital/algo/read_track_descriptor_set.h:56]: Okay, I guess we decided we're punting on Doxygen formatting for now?

File: vital/algo/read_track_descriptor_set.cxx [hunk 9]
```diff
@@ -18,25 +18,27 @@
...
+
 namespace vital {
+
 namespace algo {
 
 read_track_descriptor_set
 ::read_track_descriptor_set()
   : m_stream( 0 )
-  , m_stream_owned( false )
+    , m_stream_owned( false )
```
Comment (mwoehlke-kitware) [inline vital/algo/read_track_descriptor_set.cxx:33]: Move `,` to previous line? (Can't suggest across deletions 😢.)

**Review by mwoehlke-kitware — CHANGES_REQUESTED** [review]

File: vital/algo/video_input.cxx [hunk 10]
```diff
@@ -56,13 +66,18 @@ ::get_implementation_capabilities() const
...
 {
   m_capabilities.set_capability( name, val );
 }
 
-} } } // end namespace
+} // namespace algo
+
+} // namespace vital
+
+}     // end namespace
```
[the remark this answers is not in the document]
Comment (gbrooksby) [reply vital/algo/video_input.cxx:79]: I just grepped through and found and fixed all of these (in algo).

File: vital/algo/detected_object_set_output.cxx (hunk 3 above)
Comment (gbrooksby) [reply vital/algo/detected_object_set_output.cxx]: I grepped through and think I caught and fixed all these.

File: vital/algo/video_input.cxx [hunk 11]
```diff
@@ -7,23 +7,33 @@
...
+const algorithm_capabilities::capability_name_t video_input::HAS_FRAME_NUMBERS(
+  "has-frame-numbers" );
+const algorithm_capabilities::capability_name_t video_input::HAS_FRAME_TIME(
+  "has-frame-time" );
+const algorithm_capabilities::capability_name_t video_input::HAS_FRAME_DATA(
+  "has-frame-data" );
+const algorithm_capabilities::capability_name_t video_input::HAS_FRAME_RATE(
+  "has-frame-rate" );
+const algorithm_capabilities::capability_name_t video_input::
+HAS_ABSOLUTE_FRAME_TIME( "has-abs-frame-time" );
```
[the remark this answers is not in the document]
Comment (gbrooksby) [reply vital/algo/video_input.cxx:30]: Done.  I reformatted them all.

File: vital/algo/read_track_descriptor_set.h (hunk 8 above)
Comment (gbrooksby) [reply vital/algo/read_track_descriptor_set.h:56]: This is reformatted now.  See comment on detected_object_set_input.h above.

File: vital/algo/video_input.cxx [hunk 12]
```diff
@@ -7,23 +7,33 @@
  * \brief video_input algorithm definition instantiation
  */
 
-#include <vital/algo/video_input.h>
 #include <vital/algo/algorithm.txx>
+#include <vital/algo/video_input.h>
```
[the remark this answers is not in the document]
Comment (gbrooksby) [reply vital/algo/video_input.cxx:11]: Fixed.

File: vital/algo/estimate_fundamental_matrix.h (hunk 6 above)
Comment (gbrooksby) [reply vital/algo/estimate_fundamental_matrix.h]: Fixed.

File: vital/algo/video_input.cxx [hunk 13]
```diff
@@ -56,13 +75,18 @@ ::get_implementation_capabilities() const
 // ------------------------------------------------------------------
```
Comment (mwoehlke-kitware) [inline vital/algo/video_input.cxx:75]: We don't need to do this now, but another thing we should look into eventually is finding all of these and normalizing them to 79 characters line length. (This one is too short.)

File: vital/algo/video_input.cxx [hunk 14]
```diff
@@ -8,22 +8,41 @@
...
-const algorithm_capabilities::capability_name_t video_input::HAS_FRAME_RATE( "has-frame-rate" );
-const algorithm_capabilities::capability_name_t video_input::HAS_ABSOLUTE_FRAME_TIME( "has-abs-frame-time" );
-const algorithm_capabilities::capability_name_t video_input::HAS_METADATA( "has-metadata" );
-const algorithm_capabilities::capability_name_t video_input::HAS_TIMEOUT( "has-timeout" );
-const algorithm_capabilities::capability_name_t video_input::IS_SEEKABLE( "is-seekable" );
+const algorithm_capabilities::capability_name_t
+video_input::HAS_EOV( "has-eov" );
+const algorithm_capabilities::capability_name_t
+video_input::HAS_FRAME_NUMBERS(
+  "has-frame-numbers" );
```
Comment (mwoehlke-kitware) [inline vital/algo/video_input.cxx:25]: Hmm, it is odd that some of these are one line and some are two. I think with the type always on its own line, the remainder should always fit on a second line?

File: vital/algo/initialize_cameras_landmarks.h [hunk 15]
```diff
@@ -55,18 +65,20 @@ class VITAL_ALGO_EXPORT initialize_cameras_landmarks
...
   virtual void
-  initialize(kwiver::vital::camera_map_sptr& cameras,
-             kwiver::vital::landmark_map_sptr& landmarks,
-             kwiver::vital::feature_track_set_sptr tracks,
-             kwiver::vital::sfm_constraints_sptr constraints = nullptr) const = 0;
+  initialize( kwiver::vital::camera_map_sptr& cameras,
+              kwiver::vital::landmark_map_sptr& landmarks,
+              kwiver::vital::feature_track_set_sptr tracks,
+              kwiver::vital::sfm_constraints_sptr constraints = nullptr ) const
+  = 0;
```
Comment (mwoehlke-kitware) [inline vital/algo/initialize_cameras_landmarks.h:72]:
```suggestion
  initialize(
    kwiver::vital::camera_map_sptr& cameras,
    kwiver::vital::landmark_map_sptr& landmarks,
    kwiver::vital::feature_track_set_sptr tracks,
    kwiver::vital::sfm_constraints_sptr constraints = nullptr ) const = 0;
```

File: vital/algo/handle_descriptor_request.h [hunk 16]
```diff
@@ -30,7 +32,8 @@ class VITAL_ALGO_EXPORT handle_descriptor_request
   /// Set this algorithm's properties via a config block
   virtual void set_configuration( kwiver::vital::config_block_sptr config );
   /// Check that the algorithm's currently configuration is valid
-  virtual bool check_configuration( kwiver::vital::config_block_sptr config ) const;
+  virtual bool check_configuration( kwiver::vital::config_block_sptr config )
+  const;
```
Comment (mwoehlke-kitware) [inline vital/algo/handle_descriptor_request.h:36]:
```suggestion
  virtual bool check_configuration(
    kwiver::vital::config_block_sptr config ) const;
```

File: vital/algo/estimate_similarity_transform.cxx [hunk 17]
```diff
@@ -130,48 +137,54 @@ ::estimate_transform(camera_map_sptr const from,
...
-  std::vector<vector_3d> from_pts, to_pts;
+  std::vector< vector_3d > from_pts, to_pts;
   landmark_map::map_landmark_t from_map = from->landmarks(),
                                to_map = to->landmarks();
-  map_to_pts< landmark_map::map_landmark_t, &landmark::loc >
-    (from_map, to_map, from_pts, to_pts);
-  return this->estimate_transform(from_pts, to_pts);
+  map_to_pts< landmark_map::map_landmark_t, &landmark::loc >( from_map, to_map,
+                                                              from_pts,
+                                                              to_pts );
```
Comment (mwoehlke-kitware) [inline vital/algo/estimate_similarity_transform.cxx:180]:
```suggestion
  map_to_pts< landmark_map::map_landmark_t, &landmark::loc >(
    from_map, to_map, from_pts, to_pts );
```

File: vital/algo/estimate_similarity_transform.cxx [hunk 18]
```diff
@@ -82,46 +86,49 @@ namespace
...
+      from_pts.push_back( ( ( *from_it->second ).*afunc )() );
+      to_pts.push_back( ( ( *to_it->second ).*afunc )() );
+      ++from_it;
+      ++to_it;
     }
   }
 }
 
-}
+} // namespace
```
Comment (mwoehlke-kitware) [inline vital/algo/estimate_similarity_transform.cxx:121]:
```suggestion
} // namespace <anonymous>
```
(Yeah, not implemented in Uncrustify yet...)

File: vital/algo/read_track_descriptor_set.h (hunk 8 above)
Comment (mwoehlke-kitware) [reply vital/algo/read_track_descriptor_set.h]: FYI, there is still a mix of better and not-so-great Doxygen formatting, though again, I'm not sure we necessarily want to deal with it here.

File: vital/algo/video_input.cxx (hunk 10 above)
[the remark this answers is not in the document]
Comment (mwoehlke-kitware) [reply vital/algo/video_input.cxx]: I see you caught a bunch in files that this PR wasn't previously touching; yay!

File: vital/algo/video_input.cxx (hunk 14 above)
Comment (gbrooksby) [reply vital/algo/video_input.cxx:25]: I agree.  I've fixed these.

File: vital/algo/estimate_similarity_transform.cxx (hunk 18 above)
Comment (gbrooksby) [reply vital/algo/estimate_similarity_transform.cxx:121]: Fixed.

File: vital/algo/detected_object_set_input.h [hunk 19]
```diff
@@ -10,19 +10,22 @@
...
 
 namespace kwiver {
+
 namespace vital {
+
 namespace algo {
 
 // ----------------------------------------------------------------
+
 /**
```
[the remark this answers is not in the document]
Comment (gbrooksby) [reply vital/algo/detected_object_set_input.h:29]: This one is fixed.  Any others should get fixed in the general comment reformatting that is pending.

File: vital/algo/write_object_track_set.cxx [hunk 20]
```diff
@@ -29,8 +29,8 @@ namespace algo {
 
 write_object_track_set
 ::write_object_track_set()
-  : m_stream( 0 )
-  , m_stream_owned( false )
+  : m_stream( 0 ),
+  m_stream_owned( false )
```
[the remark this answers is not in the document]
Comment (gbrooksby) [reply vital/algo/write_object_track_set.cxx]: This one is fixed.

File: vital/algo/handle_descriptor_request.h (hunk 16 above)
Comment (gbrooksby) [reply vital/algo/handle_descriptor_request.h]: This is fixed.

File: vital/algo/estimate_similarity_transform.cxx (hunk 17 above)
Comment (gbrooksby) [reply vital/algo/estimate_similarity_transform.cxx]: This is fixed.

File: vital/algo/detected_object_set_input.h (hunk 1 above)
Comment (gbrooksby) [reply vital/algo/detected_object_set_input.h]: I'll consider this fixed for now based on the fact that _some_ kind of indention was added.

File: vital/algo/video_input.cxx (hunk 13 above)
Comment (gbrooksby) [reply vital/algo/video_input.cxx]: I fixed these in this file only.  I'll have a script standardize all the files where necessary.

**Review by gbrooksby — APPROVED** [review]
I believe all the changes requested have been addressed and this PR can be merged now.

**Pull request merged**